x-pack/test/functional/services/ml/api.ts

```diff
@@ -247,12 +247,10 @@ export function MachineLearningAPIProvider({ getService }: FtrProviderContext) {
         log.debug(`Inference endpoint '${inferenceId}' already exists. Nothing to create.`);
         return;
       }
-      const { body, status } = await esSupertest
+      const response = await esSupertest
         .put(`/_inference/${taskType}/${inferenceId}`)
         .send(requestBody);
-      this.assertResponseStatusCode(200, status, body);
-
-      return body;
+      return response;
```

Comment:
it's fine to return both body and statues, but let's keep the original assertion. 
```suggestion
      this.assertResponseStatusCode(200, response.status, response.body);
      return response;
```

Reply:
In Serverless, creating inference endpoint api may return `Request time out ` when underlying model is to be downloaded and deployed. The response code will be 408 in this case. 
From what I know the request time out doesn't occur in ESS. To handle both scenarios, I added the logic to check the status code to test suite. 

`this.assertResponseStatusCode(200, response.status, response.body);`

If we added above, I think the test suite will fail against serverless deployment. since the status code would be 408

Reply:
@saarikabhasi I'm confused. Recently @sphilipse merge a PR with special handling for request timeout, you can find it [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/server/models/model_management/models_provider.ts#L613). I presume it's no longer possible to receive 408 error.

Reply:
Fixed this. the tests were using es super user to create inference endpoint, changed to use internal path. Removed the extra check to handle request time out

Commit- https://github.com/elastic/kibana/pull/187521/commits/bc4ecbedb9e6e91a889f7df6f2ef4aa7efd7762a